internal/utils/net/ip.go

```diff
@@ -0,0 +1,50 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package net
+
+import (
+	"net"
+	"os"
+
+	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
+)
+
+const (
+	IPv4ListenerAddress = "0.0.0.0"
+	IPv6ListenerAddress = "::"
+)
+
+func IsIPv6(s string) bool {
+	ip := net.ParseIP(s)
+	if ip == nil {
+		return false
+	}
+	return ip.To4() == nil
+}
+
+// IsIPv6FirstPod returns true if the POD_IP environment variable is an IPv6 address.
+// WARNING: This function is only intended to be used in the context of Kubernetes.
+func IsIPv6FirstPod() bool {
+	return IsIPv6(os.Getenv("POD_IP"))
+}
+
+func PreferIPFamily(ipv6First bool, envoyProxy *egv1a1.EnvoyProxy) egv1a1.IPFamily {
```

Comment:
We should default to IPv4 to align with the API semantics.

```
	// IPFamily specifies the IP family for the EnvoyProxy fleet.
	// This setting only affects the Gateway listener port and does not impact
	// other aspects of the Envoy proxy configuration.
	// If not specified, the system will operate as follows:
	// - It defaults to IPv4 only.
	// - IPv6 and dual-stack environments are not supported in this default configuration.
	// Note: To enable IPv6 or dual-stack functionality, explicit configuration is required.
	// +kubebuilder:validation:Enum=IPv4;IPv6;DualStack
	// +optional
	IPFamily *IPFamily `json:"ipFamily,omitempty"`
```

Or we want to change to:

Default to the IP family of th POD IP of the Envoy Proxy?

Should we always default to IPv4 and leting users to explicitly specify IPv6 if they want?  It will make the configuration more explicit.